src/clients/api/queries/useGetTreasuryTotals.ts

```diff
@@ -0,0 +1,113 @@
+import { useMemo } from 'react';
+import BigNumber from 'bignumber.js';
+import { useGetMarkets, useGetVTokenBalancesAll, IGetVTokenBalancesAllOutput } from 'clients/api';
+import { TREASURY_ADDRESS } from 'config';
+import { VBEP_TOKENS } from 'constants/tokens';
+import { Market } from 'types';
+import { indexBy } from 'utilities/common';
+
+export interface IData {
+  treasuryTotalSupplyUsdBalanceCents: BigNumber;
+  treasuryTotalBorrowUsdBalanceCents: BigNumber;
+  treasuryTotalUsdBalanceCents: BigNumber;
+  treasuryTotalAvailableLiquidityUsdBalanceCents: BigNumber;
+}
+export interface UseGetTreasuryTotalsOutput {
+  isLoading: boolean;
+  data: IData;
+}
+
+const useGetTreasuryTotals = (): UseGetTreasuryTotalsOutput => {
+  const {
+    data: getMarketsData = {
+      markets: [],
+    },
+    isLoading: isGetMarketsLoading,
+  } = useGetMarkets({
+    placeholderData: {
+      markets: [],
+      dailyVenus: new BigNumber(0),
+    },
+  });
+
+  const vTokenAddresses: string[] = Object.values(VBEP_TOKENS).reduce(
+    (acc, item) => (item.address ? [...acc, item.address] : acc),
+    [],
+  );
+
+  const { data: vTokenBalancesTreasury = [], isLoading: isGetVTokenBalancesTreasuryLoading } =
+    useGetVTokenBalancesAll(
+      {
+        account: TREASURY_ADDRESS,
+        vTokenAddresses,
+      },
+      {
+        placeholderData: [],
+      },
+    );
+
+  const treasuryBalances = useMemo(
+    () =>
+      indexBy(
+        (item: IGetVTokenBalancesAllOutput[number]) => item.vToken.toLowerCase(), // index by vToken address
+        vTokenBalancesTreasury,
+      ),
+    [JSON.stringify(vTokenBalancesTreasury)],
+  );
+
+  const { markets } = getMarketsData;
+  const {
+    treasuryTotalSupplyUsdBalanceCents,
+    treasuryTotalBorrowUsdBalanceCents,
+    treasuryTotalUsdBalanceCents,
+    treasuryTotalAvailableLiquidityUsdBalanceCents,
+  } = useMemo(() => {
+    const data = (markets as Market[]).reduce(
```

Comment:
`markets` should already be typed

Reply:
I get errors that it isn't which is why I included the type

Reply:
updated